x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/analytics/top_metrics.yml

```diff
@@ -757,3 +757,108 @@
+          - { index: { } }
+          - { gender: 1, salary: 1000, birth_date: 1981 }
+          - { index: { } }
+          - { gender: 1, salary: 2000, birth_date: 1982 }
+          - { index: { } }
+          - { gender: 1, salary: 3000, birth_date: 1981 }
+          - { index: { } }
+          - { gender: 1, salary: 4000, birth_date: 1982 }
+          - { index: { } }
+          - { gender: 2, salary: 6000, birth_date: 1982 }
+          - { index: { } }
+          - { gender: 2, salary: 7000, birth_date: 1981 }
+          - { index: { } }
+          - { gender: 2, salary: 8000, birth_date: 1982 }
+          - { index: { } }
+          - { gender: 2, salary: 9000, birth_date: 1981 }
+
+  # Similar to a SQL query for top salaries per birth date, grouped by gender:
+  #
+  # SELECT gender, FIRST(salary, birth_date) as first
+  # FROM test_emp
+  # GROUP BY gender
+  # ORDER BY gender
+  - do:
+      search:
+        index: test
+        size: 0
+        body:
+          aggs:
+            groupby:
+              composite:
+                size: 1000
+                sources:
+                  - gender_comp:
+                      terms:
+                        field: gender
+                        missing_bucket: true
+                        order: asc
+              aggs:
+                top_salary:
+                  top_metrics:
+                    metrics:
+                      field: salary
+                    sort:
+                      birth_date: asc
+                    size: 1
+
+  - length: { aggregations.groupby.buckets: 2}
+  - match: { aggregations.groupby.buckets.0.key.gender_comp: 1}
+  - match: { aggregations.groupby.buckets.0.top_salary.top.0.sort.0: 1981}
+  - match: { aggregations.groupby.buckets.0.top_salary.top.0.metrics.salary: 1000}
+  - match: { aggregations.groupby.buckets.1.key.gender_comp: 2}
+  - match: { aggregations.groupby.buckets.1.top_salary.top.0.sort.0: 1981}
+  - match: { aggregations.groupby.buckets.1.top_salary.top.0.metrics.salary: 7000}
+
+  # Similar to a SQL query with HAVING clause:
+  #
+  # SELECT gender, FIRST(salary, birth_date) as first
+  # FROM test_emp
+  # GROUP BY gender
+  # HAVING first < 80000
+  # ORDER BY gender
+  - do:
+      search:
+        index: test
+        size: 0
+        body:
+          aggs:
+            groupby:
+              composite:
+                size: 1000
+                sources:
+                  - gender_comp:
+                      terms:
+                        field: gender
+                        missing_bucket: true
+                        order: asc
+              aggs:
+                top_salary:
+                  top_metrics:
+                    metrics:
+                      field: salary
+                    sort:
+                      birth_date: asc
+                    size: 1
+                having.top_salary:
+                  bucket_selector:
+                    buckets_path:
+                      ts: top_salary[salary]
```

Comment:
maybe also add a test where a field is selected that doesn't exist and where the top_salary bucket is empty?